### ISSUE TITLE
Quantity and numpy.array constructors behaviour change: doesn't flatten nested objects

### ISSUE THREAD

**Issue by achilleas-k** [NeuralEnsemble/python-neo#638]
A recent change in numpy and quantities is affecting the behaviour of certain common IO tests.

When creating Epochs for the common tests using the [generate_one_simple_segment](https://github.com/NeuralEnsemble/python-neo/blob/b7ab4c5db5ec1174acf80502a12b32f53aa540a9/neo/test/generate_datasets.py#L43) function, the `times` array is a nested list of single item arrays. With numpy 1.15, when `times` is used to construct a Quantity, the list is flattened. Since numpy 1.16, the behaviour seems to have changed and it now preserves the array structure and the `dtype` of the resulting array is simply `Object` (`dtype('O')`).

Consider the following script, based on [these lines that generate the Epoch for testing](
https://github.com/NeuralEnsemble/python-neo/blob/b7ab4c5db5ec1174acf80502a12b32f53aa540a9/neo/test/generate_datasets.py#L103-L118)

```python
import numpy as np
from numpy.random import rand
import quantities as pq


print("numpy version", np.__version__)

epoch_duration_range = [.5, 3.]
t = 0
times = []
durations = []
duration = 6 * pq.s
while t < duration:
    times.append(t)
    dur = rand() * np.diff(epoch_duration_range)
    dur += epoch_duration_range[0]
    durations.append(dur)
    t = t + dur

print("times list:", times)
# timesa is created the same way it's created in the Quantity() constructor
timesa = np.array(times, dtype=None, copy=True).view(pq.Quantity)
print("times array:", timesa)
print("tmesa dtype:", timesa.dtype)

# creating times quantity timesq
timesq = pq.Quantity(times, units=pq.s)
print("times quantity:", timesq)
print("timesq dtype:", timesq.dtype)
```

The output differs between numpy 1.15 and 1.16 (quantities 0.12.2):
1.15: arrays and quantities are flattened, dtype is `float64`
```
numpy version 1.15.0
times list: [0, array([1.38118117]), array([2.49708036]), array([3.01890325]), array([3.91824179]), array([5.72211357])]
times array: [0.         1.38118117 2.49708036 3.01890325 3.91824179 5.72211357] dimensionless
tmesa dtype: float64
times quantity: [0.         1.38118117 2.49708036 3.01890325 3.91824179 5.72211357] s
timesq dtype: float64
```

1.16: arrays are nested, dtype is `object`
```
numpy version 1.16.0
times list: [0, array([0.55971064]), array([2.04460238]), array([4.66616668])]
times array: [0 array([0.55971064]) array([2.04460238]) array([4.66616668])] dimensionless
tmesa dtype: object
times quantity: [0 array([0.55971064]) array([2.04460238]) array([4.66616668])] s
timesq dtype: object
```

**Comment by achilleas-k** [comment]
I dug into numpy to figure out where this came from and it looks like the old behaviour was considered a bug and the new behaviour is fixed: https://github.com/numpy/numpy/commit/c9ffde629b0f10255714b84fe75681c7f9fc4251

**Comment by samuelgarcia** [comment]
Hi @achilleas-k .
My first idea would be to fix our code.
We should have a list a scalar and not a mixed list of scalar AND 1D arrayu of size 1.
I would this:
```
dur = rand() * np.diff(epoch_duration_range)[0]
```

**Comment by apdavison** [comment]
I think I fixed this in e67c935
